### PR TITLE
[AL-65] Update account api (password/username)

### PR DESCRIPTION
**Pull request by pvnamGit**
## Jira link

[AL-65](https://capstone-alumni.atlassian.net/jira/software/c/projects/AL/boards/1?modal=detail&selectedIssue=AL-65&assignee=60cdb1028f2cc100697948aa)
## Postman (BE)
- Error when update username but don't update password
Screenshot from 2023-01-11 11-47-46

- Username is existed
Screenshot from 2023-01-11 11-48-15

- update success with user hasn't username yet
Screenshot from 2023-01-11 11-48-58

- Prevent update username
Screenshot from 2023-01-11 12-06-53

- Wrong old password
Screenshot from 2023-01-11 11-49-35

- new password 
Screenshot from 2023-01-11 11-56-10
same old password

- Update password su
Screenshot from 2023-01-11 11-59-17
ccess

## Checklist before requesting a review
1. [x] I have performed a self-review of my code
2. [x] Have you lint your code locally before submission?

### CODE REVIEW

File: src/modules/account/service/account.service.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      if (!account || user.username) {
+        throw new Error("Can't update username");
+      }
+
+      // Require update password when update username
+      if (!body.newPassword) {
+        throw new Error('Must update password');
+      }
+
+      const isUsernameExisted = await prisma.user.findFirst({
```
Comment (congtrung6391) [inline src/modules/account/service/account.service.ts:30]: could we use data from `user` instead?

File: src/modules/account/service/account.service.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      }
+
+      accountUpdateInfo = {
+        username: body.username,
+      };
+    }
+    const { password: passwordEncrypted } = user;
+    if (body.oldPassword) {
+      if (
+        !user.password ||
```
Comment (congtrung6391) [inline src/modules/account/service/account.service.ts:44]: use passwordEncrypted

File: src/pages/api/account/[id]/index.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import AccountController from 'src/modules/account/controller/account.controller';
+
+export default function accountHandler(
+  req: NextApiRequest,
+  res: NextApiResponse,
+) {
+  const { method } = req;
+  switch (method) {
+    case 'GET':
+      break;
```
Comment (congtrung6391) [inline src/pages/api/account/[id]/index.ts:12]: remove case 'GET'

File: src/modules/account/service/account.service.ts (hunk 1 above)
Comment (pvnamGit) [reply src/modules/account/service/account.service.ts:30]: How? I don't clear your question

Comment (congtrung6391) [reply src/modules/account/service/account.service.ts:30]: oh I see

File: src/pages/api/account/[id]/index.ts (hunk 3 above)
Comment (pvnamGit) [reply src/pages/api/account/[id]/index.ts]: updated

File: src/modules/account/service/account.service.ts (hunk 2 above)
Comment (pvnamGit) [reply src/modules/account/service/account.service.ts]: updated

**Pull request merged**